test/data.jl

```diff
@@ -378,4 +378,8 @@ module TestData
     df = DataFrame(x = [3, 1, 2, 1, missing], y = ["b", "c", "a", "b", "c"])
     @test_throws TypeError filter(r -> r[:x] > 1, df)
     @test_throws TypeError filter!(r -> r[:x] > 1, df)
+
+    #test corner case of getindex
```

Comment:
Sorry to do the review by parts, but shouldn't this go to dataframes.jl instead? Our tests are a bit messy, but I don't think this file tests `getindex`.

Reply:
I had a problem where to put this test as I could not track down the proper spot. In which file would you say this test belongs?

Reply:
There's something about indexing in dataframes.jl. Not sure whether there's a more appropriate place. I'm surprised we don't have systematic tests about indexing (or I missed them).

Reply:
I moved the test to `dataframes.jl`.